shared/templates/service_disabled/ansible.template

```diff
@@ -19,7 +19,7 @@
           meta: noop
 
 - name: "Unit Socket Exists - {{{ DAEMONNAME }}}.socket"
-  command: systemctl list-unit-files {{{ DAEMONNAME }}}.socket
+  command: systemctl -q list-unit-files {{{ DAEMONNAME }}}.socket | grep "{{{ DAEMONNAME }}}.socket"
```

Comment:
It is not necessary to use `grep` here. The `systemctl` command will only show the line if it exists. The return code will be `0` if the socket unit is present or `1` otherwise.

Reply:
Hello @marcusburghardt ,

On ubuntu and suse I have 
When we don't have a socket:
 
u001@rchost:~$ systemctl list-unit-files time-sync.socket 
UNIT FILE STATE VENDOR PRESET

0 unit files listed.

When we have a socket
u001@rchost:~$ systemctl list-unit-files libvirtd.socket 
UNIT FILE       STATE   VENDOR PRESET
libvirtd.socket enabled enabled      

1 unit files listed.

Whit grep in the first case we have 

u001@rchost:~$  systemctl -q list-unit-files time-sync.socket | grep time-sync.socket
u001@rchost:~$ 


Whit grep in the second case case we have 
u001@rchost:~$ systemctl -q list-unit-files libvirtd.socket | grep  libvirtd.socket
libvirtd.socket enabled enabled  

The rule as exist on SUSE does not work. 
Have a nice day
Rumen

Reply:
I see, but grep is still not necessary. See this example:
```
$ systemctl -q list-unit-files sshd.socket ; echo $?
sshd.socket disabled disabled
0

$ systemctl -q list-unit-files absent.socket ; echo $?
1
```
You can simplify the command and the conditional which uses its return code.

Reply:
Hello @marcusburghardt 
It does not work as expected. I have attached a screen shot taken from my console - Ubuntu 20.04. I think that the command systemctl was changed and after its change we have an additional message "UNIT FILE STATE VENDOR PRESET". Please try on another OS.  
Have a nice day
Rumen
![socket_issue](https://github.com/ComplianceAsCode/content/assets/77793453/4dcd116d-695b-4b03-9971-98f0b4fcbe5a)

Reply:
Ok, this change in the systemctl return codes for Ubuntu is new for me.
Even that, since you are using the `search` for the conditional below, it is still not necessary to use `grep` here because the output of `systemctl` command will also include the unit name if it is present.

Reply:
Hello @marcusburghardt 

This way of working is not only for Ubuntu, but for SUSE as well (Note: SLE 15 SP 2 - works as before, but SLE 15 SP4 works in the new way). Only via grep I was able to get an empty string when socket does not exist or the name of the socket with its status when the socket exist. You can see that the usage of echo does not remove this "parasite row" - "UNIT FILE...."  i.e I did not have a "plain vanilla" return code. May you check how this works on RedHat - different versions - old ones and a new one?

I can add if / else condition which check the product and to use the proposed syntax with grep for Ubuntu and SUSE. 

Have a nice day
Rumen